main/src/test/java/org/mobilitydata/gtfsvalidator/validator/TimepointTimeValidatorTest.java

```diff
@@ -17,110 +17,356 @@
+            .build());
+    stopTimes.add(
+        new GtfsStopTime.Builder()
+            .setCsvRowNumber(1)
+            .setTripId("second trip id")
+            .setArrivalTime(GtfsTime.fromSecondsSinceMidnight(450))
+            .setDepartureTime(GtfsTime.fromSecondsSinceMidnight(580))
+            .setStopId("stop id 2")
+            .setStopSequence(2)
+            .setTimepoint(1)
+            .build());
+    assertThat(generateNotices(createHeaderWithTimepointColumn(), stopTimes)).isEmpty();
+  }
+
+  @Test
+  public void emptyTimepoint_noTimesProvided_shouldGenerateNotice() {
+    List<GtfsStopTime> stopTimes = new ArrayList<>();
+    stopTimes.add(
+        new GtfsStopTime.Builder()
+            .setCsvRowNumber(1)
+            .setTripId("first trip id")
+            .setArrivalTime(null)
+            .setDepartureTime(null)
+            .setStopId("stop id")
+            .setStopSequence(2)
+            .setTimepoint((Integer) null)
+            .build());
+    stopTimes.add(
+        new GtfsStopTime.Builder()
+            .setCsvRowNumber(1)
+            .setTripId("second trip id")
+            .setArrivalTime(GtfsTime.fromSecondsSinceMidnight(450))
+            .setDepartureTime(GtfsTime.fromSecondsSinceMidnight(580))
+            .setStopId("stop id 2")
+            .setStopSequence(2)
+            .setTimepoint(1)
+            .build());
+
+    assertThat(generateNotices(createHeaderWithTimepointColumn(), stopTimes))
+        .containsExactly(new MissingTimepointValueNotice(stopTimes.get(0)));
   }
 
   @Test
-  public void nonTimepointShouldNotGenerateNotice() {
-    assertThat(
-            generateNotices(
-                new GtfsStopTime.Builder()
-                    .setCsvRowNumber(1)
-                    .setTripId("first trip id")
-                    .setArrivalTime(null)
-                    .setDepartureTime(GtfsTime.fromSecondsSinceMidnight(580))
-                    .setStopId("stop id")
-                    .setStopSequence(2)
-                    .setTimepoint(0)
-                    .build()))
-        .isEmpty();
+  public void emptyTimepoint_timesProvided_shouldGenerateNotice() {
+    List<GtfsStopTime> stopTimes = new ArrayList<>();
+    stopTimes.add(
+        new GtfsStopTime.Builder()
+            .setCsvRowNumber(1)
+            .setTripId("first trip id")
+            .setArrivalTime(GtfsTime.fromSecondsSinceMidnight(450))
+            .setDepartureTime(GtfsTime.fromSecondsSinceMidnight(580))
+            .setStopId("stop id")
+            .setStopSequence(2)
+            .setTimepoint((Integer) null)
+            .build());
+    stopTimes.add(
+        new GtfsStopTime.Builder()
+            .setCsvRowNumber(1)
+            .setTripId("second trip id")
+            .setArrivalTime(GtfsTime.fromSecondsSinceMidnight(450))
+            .setDepartureTime(GtfsTime.fromSecondsSinceMidnight(580))
+            .setStopId("stop id 2")
+            .setStopSequence(2)
+            .setTimepoint(1)
```

Comment:
Based on method name doesn't seem like this StopTime record should exist in this test